### PR TITLE
Add WNP for Firefox 115 for NA [fix #13277]

### LINKED ISSUE #13277: WNP 115 for NA
## Description

Placeholder till all details are available.

[Designs](https://www.figma.com/file/CZvzHgU9NQMQUm34lomIl5/Firefox-WNP?type=design&t=BNElYnTv7WTDKAwE-0) (final)

Copy: pull from figma file

Behavior: For users on Windows 7 or older, show first version. For everyone else, show second version

### PR DESCRIPTION
**Pull request by craigcook**
## One-line summary

English WNP for Firefox 115.0, releasing on **4 July 2023**.

People using Windows 7 or older should be redirected to the variant page, all others should get the regular page promoting addons.

## Issue / Bugzilla link

#13277 

## Testing

Demo server URL: https://www-demo2.allizom.org/en-US/firefox/115.0/whatsnew/

To test this work:

http://localhost:8000/en-US/firefox/115.0/whatsnew/

Test in Firefox on Windows 7 (you can use Browserstack)
- [x] With no query string you should be redirected to `?xv=windows` and load the Win7 page.
- [x] If a query string is already present (`?foo=bar` or `?oldversion=114.0` etc) it should append `&xv=windows` and load the Win7 page.

Test in Firefox on Windows 8 or higher:
- [x] You should get the addons page.

Any non-Firefox browser or any non-Windows platform should get the addons page.

I know my JavaScript is terrible.

### CODE REVIEW

**Comment by craigcook** [comment]
Note: https://github.com/mozilla/bedrock/pull/13311 adds the WNP for EU and which ever merges first the other will need a rebase.

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx115-eu-vpn.html [hunk 1]
```diff
@@ -21,6 +21,8 @@
 {% endblock %}
 
 {% block experiments %}
+  {{ js_bundle('firefox_whatsnew_115_na_win7') }}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx115-eu-vpn.html:24]: Would this try to redirect win7 users who are viewing on a non-English locale? Should it be limited to `en` only?

File: media/js/firefox/whatsnew/whatsnew-115-na-win7.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+// Create namespace
+if (typeof window.Mozilla === 'undefined') {
+    window.Mozilla = {};
+}
```
Comment (alexgibson) [inline media/js/firefox/whatsnew/whatsnew-115-na-win7.js:10]: Nit: this doesn't seem to be needed?

File: media/js/firefox/whatsnew/whatsnew-115-na-win7.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        // re-insert hash (if originally present)
+        if (hash) {
+            redirect += '#' + hash;
+        }
+    }
+
+    if (
+        window.site.platform === 'windows' &&
+        window.site.platformVersion <= 6.1
+    ) {
```
Comment (alexgibson) [inline media/js/firefox/whatsnew/whatsnew-115-na-win7.js:41]: Perhaps add a check that `redirect` is defined here before redirecting, just to be safe?
```suggestion
    if (
        redirect &&
        window.site.platform === 'windows' &&
        window.site.platformVersion <= 6.1
    ) {
```

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx115-eu-vpn.html (hunk 1 above)
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx115-eu-vpn.html:24]: Is it worth also putting this behind a switch, incase we need to turn it off should something go awry?

**Review by alexgibson — CHANGES_REQUESTED** [review]
Tested in browserstack on demo and the redirect seems to work well for me. Nice job!

A couple of minor issues / suggestion below, but otherwise r+wc 🪟

File: media/js/firefox/whatsnew/whatsnew-115-na-win7.js (hunk 2 above)
Comment (craigcook) [reply media/js/firefox/whatsnew/whatsnew-115-na-win7.js:10]: you're right, a leftover from a previous iteration.

**Review by alexgibson — APPROVED** [review]
r+

**Review by alexgibson — APPROVED** [review]
r++

**Pull request merged**